cinema/views.py

```diff
@@ -1 +1,71 @@
-# write views here
+from rest_framework import viewsets
+
+from cinema.models import Movie, MovieSession, Genre, Actor, CinemaHall
+from cinema.serializers import (
+    MovieSerializer,
+    MovieListSerializer,
+    MovieDetailSerializer,
+    MovieSessionListSerializer,
+    MovieSessionDetailSerializer,
+    MovieSessionSerializer,
+    GenreSerializer,
+    ActorSerializer,
+    CinemaHallSerializer
+)
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
+    serializer_class = MovieSerializer
+
+    def get_queryset(self):
+        if self.action == "list":
+            return self.queryset.prefetch_related("genres", "actors")
+        return self.queryset
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieListSerializer
+
+        if self.action == "retrieve":
+            return MovieDetailSerializer
+
+        return MovieSerializer
```

Comment:
Call `self.serializer_class` here since it is defined in the view. Apply this change in all places
```suggestion
        return self.serializer_class
```